pkg/gardenlet/operation/botanist/advertisedaddresses_test.go

```diff
@@ -183,6 +186,47 @@ var _ = Describe("AdvertisedAddresses", func() {
 			}))
 		})
 
+		It("returns external, internal addresses with addition to default service-account-issuer address because ShootManagedIssuer is disabled", func() {
+			DeferCleanup(test.WithFeatureGate(features.DefaultFeatureGate, features.ShootManagedIssuer, false))
+			botanist.Shoot.ExternalClusterDomain = ptr.To("foo.bar")
+			botanist.Shoot.InternalClusterDomain = "baz.foo"
+			botanist.Shoot.ServiceAccountIssuerHostname = ptr.To("managed.foo.bar")
+			botanist.Garden = &garden.Garden{
+				Project: &gardencorev1beta1.Project{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "some-proj",
+					},
+				},
+			}
+
+			botanist.Shoot.GetInfo().ObjectMeta = metav1.ObjectMeta{
+				Name:      "test",
+				Namespace: "testspace",
+				UID:       "some-uid",
+				Annotations: map[string]string{
+					"authentication.gardener.cloud/issuer": "managed",
+				},
+			}
+			botanist.APIServerAddress = "bar.foo"
+
+			addresses, err := botanist.ToAdvertisedAddresses()
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(addresses).To(Equal([]gardencorev1beta1.ShootAdvertisedAddress{
+				{
+					Name: "external",
+					URL:  "https://api.foo.bar",
+				}, {
+					Name: "internal",
+					URL:  "https://api.baz.foo",
+				},
+				{
+					Name: "service-account-issuer",
+					URL:  "https://api.baz.foo",
+				},
+			}))
+		})
```

Comment:
I see no changes in the `ToAdvertisedAddresses` function, can you explain how this unit test is relevant/testing anything? 👀

Reply:
The function `ToAdvertisedAddresses` always reports the `service-account-issuer` address if possible, not only if the managed issuer is enabled. Then it calls `computeKubeAPIServerServiceAccountConfig` which uses the feature gate to determine if the annotation should be taken into consideration. The feature gate is set to `false` in the test and this is the reason why the `service-account-issuer` address is equal to the default `internal` address.